src/prefect/projects/base.py

```diff
@@ -0,0 +1,226 @@
+            shell=sys.platform == "win32",
+            stderr=subprocess.DEVNULL,
+        )
+        repository = p.decode().strip()
+        is_git_based = True
+        name = name or "/".join(p.decode().strip().split("/")[-2:]).replace(".git", "")
+    except subprocess.CalledProcessError:
+        dir_name = os.path.basename(os.getcwd())
+        name = name or dir_name
+
+    # hand craft a pull step
+    if is_git_based:
+        try:
+            p = subprocess.check_output(
+                ["git", "rev-parse", "--abbrev-ref", "HEAD"],
+                shell=sys.platform == "win32",
+                stderr=subprocess.DEVNULL,
+            )
+            branch = p.decode().strip()
+        except subprocess.CalledProcessError:
+            branch = "main"
+
+        pull_step = [
+            {
+                "prefect.projects.steps.git_clone_project": {
+                    "repository": repository,
+                    "branch": branch,
+                }
+            }
+        ]
+    else:
+        pull_step = [
+            {
+                "prefect.projects.steps.set_working_directory": {
+                    "directory": str(Path(".").absolute().resolve()),
+                }
+            }
+        ]
+
+    files = []
+    if set_default_ignore_file("."):
+        files.append(".prefectignore")
+    if set_default_deployment_yaml("."):
+        files.append("deployment.yaml")
+    if set_default_project_yaml(".", name=name, pull_step=pull_step):
+        files.append("prefect.yaml")
+    if set_prefect_hidden_dir():
+        files.append(".prefect/")
+
+    return files
+
+
+async def register_flow(entrypoint: str, force: bool = False):
+    """
+    Register a flow with this project from an entrypoint.
+
+    Raises a ValueError if registration will overwrite an existing known flow.
+    """
+    try:
+        fpath, obj_name = entrypoint.rsplit(":", 1)
+    except ValueError as exc:
+        if str(exc) == "not enough values to unpack (expected 2, got 1)":
+            missing_flow_name_msg = (
+                "Your flow entrypoint must include the name of the function that is"
+                f" the entrypoint to your flow.\nTry {entrypoint}:<flow_name>"
+            )
+            raise ValueError(missing_flow_name_msg)
+        else:
+            raise exc
+
+    flow = await run_sync_in_worker_thread(load_flow_from_entrypoint, entrypoint)
+
+    fpath = Path(fpath).absolute()
+    prefect_dir = find_prefect_directory()
+    if not prefect_dir:
+        raise FileNotFoundError(
+            "No .prefect directory could be found - run `prefect project"
+            " init` to create one."
+        )
+
+    entrypoint = f"{fpath.relative_to(prefect_dir.parent)!s}:{obj_name}"
+
+    if (prefect_dir / "flows.json").exists():
+        with open(prefect_dir / "flows.json", "r") as f:
+            flows = json.load(f)
+    else:
+        flows = {}
+
+    ## quality control
+    if flows.get(flow.name) != entrypoint:
```

Comment:
We need to check if the flow is already in `flows`; otherwise, this function will raise if the flow hasn't been added to `flows.json` yet.
```suggestion
    if flow.name in flows and flows[flow.name] != entrypoint:
```

Reply:
Along a similar line, if I try to create a deployment I get the following error:

```
repeatedly get 
File"/Users/jennifer/Desktop/Prefect/prefect/src/prefect/projects/base.py", line 218, in register_flow
    "Conflicting entry found for flow with name"
KeyError: 'hi_results'
```

If I comment out lines 215-220, it works fine.